horovod/torch/mpi_ops.cc

```diff
@@ -216,6 +216,38 @@ int DoBroadcastCudaOnCPU(TC* tensor, TC* output, int root_rank, char* name) {
 }
 #endif
 
+void WaitAndClear(int handle) {
+  while (!handle_manager.PollHandle(handle)) {
+    std::this_thread::sleep_for(std::chrono::milliseconds(1));
+  }
+  auto status = handle_manager.ReleaseHandle(handle);
+  ThrowIfError(*status);
+}
+
+int DoJoin(int device) {
+  ThrowIfError(common::CheckInitialized());
+
+  auto handle = handle_manager.AllocateHandle();
+  auto ready_event = RecordReadyEvent(device);
+
+  auto hvd_cpu_buffer =
+      std::make_shared<TorchTemporaryBuffer<DataType::HOROVOD_INT64, DeviceType::CPU, THLongTensor>>(
+          CPU_DEVICE_ID);
```

Comment:
Why `CPU_DEVICE_ID`?  If we use the `device` passed in to the function, we can allocate Tensors on the correct device through the framework itself instead of needing to do it ourselves.